### PR TITLE
Capybara testing

### PR DESCRIPTION
**Pull request by AdnanOlarmmi**
Integration specs
Use Capybara to write integration tests for each view in your project.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.
Be sure to include integration specs for any other features you may have implemented!

- N+1 Problem

- Former
not working

- Present
working

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Adnan, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦  🏃 

## Highlights 👍  
- _N + 1_ problem addressed!
- Professional ReadMe!

## Required Changes ♻️
- [ ] A number of your spec cases are failing. Please ensure that all test cases are passing:

**Failing tests:**
image

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile [hunk 1]
```diff
@@ -72,7 +72,10 @@ end
 # end
 
 group :development, :test do
+  gem 'capybara'
```
Comment (rustam-tolipov) [inline Gemfile:75]: **You both did great job!** 

- [ ] However, as the previous reviewer suggested we have failing tests. Kindly, fix them before submitting the. Maybe check your db/seeds.rb file and add some preload users, posts, comments, likes... I can see you did added but the issue is still appearing. This is a video referace kindly watch it!


https://user-images.githubusercontent.com/69011963/216124154-cdd459bd-5bec-4c83-8649-0088b5a0e3ec.mov

_Try to recloen your project and reset your database then test your code. I think with this try you will see the errors._

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 1 above)
Comment (AdnanOlarmmi) [reply Gemfile:75]: @Rustamxon7 I'm so sorry, I didn't push the changes

**Review by alexander16108 — APPROVED** [review]
Hi @AdnanOlarmmi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- All linters are passing ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**